File: web/src/layout/LoadImageDemoXml.ts
//
// LoadImageDemoXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId, startupAutoResize } from 'butterfly-web/dist/views/html'
import { customViewSetDelegate } from 'butterfly-web/dist/views/CustomView'
import htmlForDefault from './load_image_demo.html'
//! Declares com.lightningkite.butterflytemplate.layouts.LoadImageDemoXml
export class LoadImageDemoXml {
    xmlRoot!: HTMLElement
    gallery!: HTMLButtonElement
    galleryMultiple!: HTMLButtonElement
    camera!: HTMLButtonElement
    loremPixel!: HTMLButtonElement
    image!: HTMLImageElement
    checkCanUpload!: HTMLButtonElement
    canUpload!: HTMLElement
    loadHtmlString(): string {
        return htmlForDefault
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString())
        this.xmlRoot = view
        this.gallery = getViewById<HTMLButtonElement>(view, "gallery")
        this.galleryMultiple = getViewById<HTMLButtonElement>(view, "galleryMultiple")
        this.camera = getViewById<HTMLButtonElement>(view, "camera")
        this.loremPixel = getViewById<HTMLButtonElement>(view, "lorem_pixel")
        this.image = getViewById<HTMLImageElement>(view, "image")
        this.checkCanUpload = getViewById<HTMLButtonElement>(view, "checkCanUpload")
        this.canUpload = getViewById<HTMLElement>(view, "canUpload")
        return view
    }
}

File: android/src/main/java/com/lightningkite/khrysalistemplate/layouts/LoadImageDemoXml.kt
//
// LoadImageDemoXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.khrysalistemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.khrysalis.views.*
import com.lightningkite.khrysalistemplate.R

class LoadImageDemoXml {

    lateinit var gallery: Button
    lateinit var galleryMultiple: Button
    lateinit var camera: Button
    lateinit var loremPixel: Button
    lateinit var image: ImageView
    lateinit var checkCanUpload: Button
    lateinit var canUpload: TextView
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ViewDependency): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.load_image_demo, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        gallery = view.findViewById<Button>(R.id.gallery)
        galleryMultiple = view.findViewById<Button>(R.id.galleryMultiple)
        camera = view.findViewById<Button>(R.id.camera)
        loremPixel = view.findViewById<Button>(R.id.lorem_pixel)
        image = view.findViewById<ImageView>(R.id.image)
        checkCanUpload = view.findViewById<Button>(R.id.checkCanUpload)
        canUpload = view.findViewById<TextView>(R.id.canUpload)
        
        
        
        return view
    }
}